tests/scripts/run-throughput-perf.py

```diff
@@ -0,0 +1,361 @@
+        crossgen_path: the path to crossgen
+        jit_path: the path to the jit
+        assemblies_path: the path to the assemblies that may be needed for the crossgen run
+    """
+
+    dll_runtimes = []
+
+    # Set up arguments for running crossgen
+    run_args = [crossgen_path,
+            '/JITPath',
+            jit_path,
+            '/Platform_Assemblies_Paths',
+            assemblies_path,
+            dll_path
+            ]
+
+    log(" ".join(run_args))
+
+    # Time.clock() returns seconds, with a resolution of 0.4 microseconds, so multiply by the multiplier to get milliseconds
+    multiplier = 1000
+
+    for iteration in range(0,iterations):
+        proc = subprocess.Popen(run_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+
+        start_time = time.clock()
+        (out, err) = proc.communicate()
+        end_time = time.clock()
+
+        if proc.returncode == 0:
+            # Calculate the runtime
+            runtime = end_time * multiplier - start_time * multiplier
+            dll_runtimes.append(runtime)
+        else:
+            log("Error in %s" % (dll_name))
+            log(err.decode("utf-8"))
+
+    return dll_runtimes
+
+##########################################################################
+# Main
+##########################################################################
+
+def main(args):
+    global dll_list
+
+    architecture, operating_system, build_type, run_type, clr_root, assembly_root, benchview_path = validate_args(args)
+    arch = architecture
+
+    current_dir = os.getcwd()
+    jit = "clrjit.dll"
+
+    # jit32 uses compatjit.dll
+    if architecture == 'x86jit32':
+        arch = 'x86'
+        jit = "compatjit.dll"
+
+    # Make sandbox
+    sandbox_path = os.path.join(clr_root, "sandbox")
+    if os.path.isdir(sandbox_path):
+        shutil.rmtree(sandbox_path, onerror=del_rw)
+
+    os.makedirs(sandbox_path)
+    os.chdir(sandbox_path)
+
+    # Set up paths
+    test_root = os.path.join(clr_root, 'bin', 'Tests', operating_system + '.' + arch + '.' + build_type)
+    core_root_path = os.path.join(test_root, 'Tests', 'Core_Root')
+
+    crossgen_path = os.path.join(core_root_path,'crossgen.exe')
+    jit_path = os.path.join(core_root_path, jit)
+
+    runtimes = {}
+    iterations = 5
+
+    # Run throughput testing
+    for dll_name in dll_list:
+        dll_file_name = dll_name + ".dll"
+        dll_path = os.path.join(assembly_root, dll_file_name)
+        dll_runtimes = runIterations(dll_file_name, dll_path, iterations, crossgen_path, jit_path, assembly_root)
+
+        if len(dll_runtimes) != 0:
+            runtimes[dll_name] = dll_runtimes
+
+    # Upload the data
+    if not benchview_path is None:
+        for dll_name, dll_runtimes in runtimes.items():
+            # Generate the csv file
+            csv_file_name = generateCSV(dll_name, dll_runtimes)
+
+            # For each benchmark, call measurement.py
+            measurement_args = ["py",
```

Comment:
There is currently a bug in the display part of Benchview around the geomean.  If you do not upload all of the data in a subtree at once we will not correctly calculate the geomean, and since we do this at insertion time and not display time you get the wrong geomean.  This should be easy to fix, by just moving the submission.json creation out of the loop along with the upload.